files/scripts/syncd.sh

```diff
@@ -44,6 +44,8 @@ function startplatform() {
         /usr/bin/mlnx-fw-upgrade.sh -v
         if [[ "$?" -ne "${EXIT_SUCCESS}" ]]; then
             debug "Failed to upgrade fw. " "$?" "Restart syncd"
+            sonic-db-cli STATE_DB HSET "FEATURE|$DEV_SRV" fail_reason \
```

Comment:
Why are we considering just the asic firmware update as sysready indication for syncd. Shouldn't it be the create switch success?